include/swift/AST/DiagnosticsSema.def

```diff
@@ -1082,13 +1082,19 @@ NOTE(unwrap_with_guard,none,
 ERROR(optional_base_not_unwrapped,none,
       "value of optional type %0 must be unwrapped to refer to member %1 of "
       "wrapped base type %2", (Type, DeclNameRef, Type))
+ERROR(invalid_optional_infered_keypath_root, none,
+      "key path root inferred as optional type %0 must be chained to refer to member %1 "
+      "of unwrapped type %2", (Type, DeclNameRef, Type))
 NOTE(optional_base_chain,none,
      "chain the optional using '?' to access member %0 only for non-'nil' "
      "base values", (DeclNameRef))
 NOTE(optional_base_remove_optional_for_keypath_root, none,
      "use unwrapped type %0 as key path root", (Type))
 NOTE(optional_keypath_application_base, none,
      "use '?' to access key path subscript only for non-'nil' base values", ())
+NOTE(optional_key_path_root_base_chain,none,
+     "chain the optional using '?.' to access unwrapped type member %0",
```

Comment:
Maybe “chain the optional using “?.” to access member %0 only for non-'nil' base values“ to be consistent with existing diagnostics?

Reply:
Not really sure about that, in fact not phrase it like that was intentional because when talking about key path the chain is to refer to `Wrapped` type members not really value on a base type optional type...